packages/checkout-storefront/src/sections/DeliveryMethods/DeliveryMethods.tsx

```diff
@@ -0,0 +1,140 @@
+import { Title } from "@/checkout-storefront/components/Title";
+import { Text } from "@saleor/ui-kit";
+import {
+  CountryCode,
+  ShippingMethod,
+  useCheckoutDeliveryMethodUpdateMutation,
+} from "@/checkout-storefront/graphql";
+import { useCheckout } from "@/checkout-storefront/hooks/useCheckout";
+import React, { useEffect, useRef, useState } from "react";
+import { useFormattedMessages } from "@/checkout-storefront/hooks/useFormattedMessages";
+import { SelectBox } from "@/checkout-storefront/components/SelectBox";
+import { SelectBoxGroup } from "@/checkout-storefront/components/SelectBoxGroup";
+import { useAlerts } from "@/checkout-storefront/hooks/useAlerts";
+import { extractMutationErrors, getById, getFormattedMoney } from "@/checkout-storefront/lib/utils";
+import { Divider } from "@/checkout-storefront/components/Divider";
+import { CommonSectionProps } from "@/checkout-storefront/lib/globalTypes";
+
+export const DeliveryMethods: React.FC<CommonSectionProps> = ({ collapsed }) => {
+  const formatMessage = useFormattedMessages();
+  const { checkout } = useCheckout();
+  const { shippingMethods, shippingAddress, deliveryMethod } = checkout;
+  const { showErrors } = useAlerts("checkoutDeliveryMethodUpdate");
+  const [selectedMethodId, setSelectedMethodId] = useState(checkout?.deliveryMethod?.id);
+  const shippingCountryRef = useRef<CountryCode | undefined | null>(
+    shippingAddress?.country?.code as CountryCode
+  );
+
+  const [, updateDeliveryMethod] = useCheckoutDeliveryMethodUpdateMutation();
+
+  const hasValidMethodSelected =
+    selectedMethodId && shippingMethods.some(getById(selectedMethodId));
+
+  const handleAutoSetMethod = () => {
+    const cheapestMethod = shippingMethods.reduce(
+      (resultMethod, currentMethod) =>
+        currentMethod.price.amount < resultMethod.price.amount ? currentMethod : resultMethod,
+      shippingMethods[0] as ShippingMethod
+    );
+
+    void handleSubmit(cheapestMethod.id);
+  };
+
+  const handleAutoSetMethodAfterMethodsListChange = () => {
+    if (hasValidMethodSelected || !shippingMethods.length) {
+      return;
+    }
+
+    handleAutoSetMethod();
+  };
+
+  useEffect(handleAutoSetMethodAfterMethodsListChange, [shippingMethods]);
```

Comment:
Missing deps. It would be easier if these functions were inlined in useEffect:

```suggestion
  useEffect(() => {
    if (hasValidMethodSelected || !shippingMethods.length) {
      return;
    }

    handleAutoSetMethod();
  }, [hasValidMethodSelected, shippingMethods, handleAutoSetMethod]);
```